rfcs/0001-extended-lookup-tables.md

```diff
@@ -0,0 +1,503 @@
+entries in the table to the full width `X` in the same way as the fixed tables
+above.
+
+We represent the masked values in the runtime table as `RuntimeTable(i)` for
+`1 <= i < X`.
+
+To specify whether a runtime table entry is applicable in the `i`th row, we use
+a selector polynomial `RuntimeTableSelector`. In order to reduce the polynomial
+degree of later checks, we add a constraint to check that the runtime entry is
+the zero vector wherever the selector is zero:
+```
+RuntimeTableSelector * sum[i=1, X, RuntimeTable(i) * joint_combiner^i]
+ = sum[i=1, X, RuntimeTable(i) * joint_combiner^i]
+```
+
+This gives the combined entry of the fixed and runtime tables as
+```
+sum[i=0, W, FixedTable(i) * joint_combiner^i]
++ sum[i=1, X, RuntimeTable(i) * joint_combiner^i]
++ TableID * joint_combiner^max_joint_size
+```
+where we assume that `RuntimeTableSelector * FixedTable(i) = 0` for
+`1 <= i < W` via our simplifying assumption above. We compute this as a
+polynomial `FixedAndRuntimeTable`, giving its evaluations and openings as part
+of the proof, and asserting that each entry matches the above.
+
+The `RuntimeTableSelector` polynomial will be fixed at constraint system time,
+and we can avoid opening it. However, we must provide openings for
+`RuntimeTable(i)`.
+
+The `joint_combiner` should be sampled using randomness that depends on the
+individual `RuntimeTable(i)`s, to ensure that the runtime values cannot be
+chosen with prior knowledge `joint_combiner`.
+
+### Representing the side-loaded tables
+
+The side-loaded tables can be considered as additional 'masks' that we apply to
+the combined fixed and runtime table, to introduce this side-loaded data.
+
+Again, we make the simplifying assumption that an entry from a side-loaded
+table has exactly 1 'fixed' entry, which can be used for indexing (or set to
+`0` if unused). We also assume that at most 1 side-loaded table contributes to
+each entry.
+
+In order to compute the lookup multiset inclusion argument, we also need the
+combined table of all 3 table kinds, which we will call `LookupTable`. We can
+calculate the contribution of the side-loaded table for each lookup as
+```
+LookupTable - FixedAndRuntimeTable
+```
+
+We will represent the polynomials for the side-loaded tables as
+`SideLoadedTable(i, j)`, where `i` identifies the table, and `j` identifies the
+column within that table. We also include polynomials `SideLoadedCombined(i)`
+in the proof, along with their openings, where
+```
+SideLoadedCombined(i)
+  = sum[j=1, ?, SideLoadedTable(i, j) * joint_combiner^(j-1)]
+```
+and check for consistency of the evaluations and openings when verifying the
+proof.
+
+The `SideLoadedTable` polynomials are included in the proof, but we do so 'as
+if' they had been part of the verification index (ie. without any
+proof-specific openings), so that they may be reused across multiple different
+proofs without modification.
```

Comment:
Would be nice to have a note about the size of this extra proof data.  For example, as a function of number of tables, max table size, etc?  also how many extra commitments and evaluations this adds could be useful

Reply:
This is explicitly laid out in the section `Full list of polynomials and constraints` below.